make/common/native/Flags.gmk

```diff
@@ -122,7 +122,7 @@ define SetupCompilerFlags
     $1_EXTRA_CXXFLAGS += $$($1_CXXFLAGS_$(OPENJDK_TARGET_OS)_release)
   endif
   ifeq ($(STATIC_LIBS), true)
-    $1_EXTRA_CXXFLAGS += $$(STATIC_LIB_CFLAGS)
+    $1_EXTRA_CXXFLAGS += -DSTATIC_BUILD=1
```

Comment:
This seems unrelated to the fix?

Reply:
> 3\. Fixed `STATIC_LIB_CFLAGS` in `Flags.gmk` to `-DSTATIC_BUILD=1`, which was missed by [cbab40bc](https://github.com/openjdk/jdk/commit/cbab40bce45a2f58906be49c841178fa1dfd457e#diff-ab3ce05e795360030f19402fd0c2fad1dc1f7c5e7acc993cc4a2096cf31ccf40R114-R121) for the refactor of building static libs. 

As described in the No.3 bullet of this PR. It showed up when I tested the EXTRA_CXXFLAGS passed from outside of `SetupNativeCompilation`. It is a simple and tiny fix which may not require a separate PR I think.

FYI, the failures on linux-x64-static and linux-x64-static-libs tests, captured by OpenJDK GHA Sanity Checks: https://github.com/cnqpzhang/jdk/actions/runs/13943096440.

Reply:
You are correct that this is a bug, but the fix should not be sneaked in under the guise of another problem.

I created https://bugs.openjdk.org/browse/JDK-8353272 for this.

Reply:
Reverted the fix. Thanks for helping file a new JBS and follow-ups.